packages/framework/tests/Feature/Actions/SeedsPublicationFilesTest.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Hyde\Framework\Testing\Feature\Actions;
+
+use function array_merge;
+use function explode;
+use function file_get_contents;
+use Hyde\Framework\Actions\SeedsPublicationFiles;
+use Hyde\Framework\Features\Publications\Models\PublicationFieldType;
+use Hyde\Framework\Features\Publications\Models\PublicationType;
+use Hyde\Hyde;
+use Hyde\Testing\TestCase;
+use PHPUnit\Framework\ExpectationFailedException;
+use function str_ends_with;
+use function str_replace;
+
+/**
+ * @covers \Hyde\Framework\Actions\SeedsPublicationFiles
+ */
+class SeedsPublicationFilesTest extends TestCase
+{
+    protected PublicationType $pubType;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->directory('test-publication');
+        $this->setupTestPublication();
+        $this->pubType = PublicationType::get('test-publication');
+    }
+
+    public function testCreate()
+    {
+        $action = new SeedsPublicationFiles($this->pubType);
+        $action->create();
+
+        $this->assertFileExists($this->firstPublicationFilePath());
+    }
+
+    public function testWithStringType()
+    {
+        $this->updateSchema('string', 'title');
+        (new SeedsPublicationFiles($this->pubType))->create();
+
+        $this->assertFileMatchesString(
+            '---
+__createdAt: ***
+title: ***
+---
+
+## Write something awesome.
+
+', $this->firstPublicationFilePath()
+        );
+    }
+
+    public function testWithArrayType()
+    {
+        $this->addToSchema('array', 'tags');
+        (new SeedsPublicationFiles($this->pubType))->create();
+
+        $this->assertFileMatchesString(
+            '---
+__createdAt: ***
+title: ***
+tags:
+  - ***
+  - ***
+  - ***
+', $this->firstPublicationFilePath(), 6);
+    }
+
+    protected function getPublicationFiles(): array
+    {
+        $files = glob(Hyde::path('test-publication/*.md'));
+        $this->assertNotEmpty($files, 'No publication files found.');
+
+        return $files;
+    }
+
+    protected function firstPublicationFilePath(): string
+    {
+        return $this->getPublicationFiles()[0];
+    }
+
+    protected function assertFileMatchesString(string $expected, string $filepath, ?int $limit = null)
```

Comment:
It might actually be easier to just parse the file as a simple markdown document object and check the matter through that.